### PR TITLE
Webpublisher: Skip version collect

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Added `CollectSceneVersion` to Settings. Added option to configure skipping it for headless publishing.

## Description
One might want to skip collecting/validating presence of version number in published workfile (in case of studio processing).
In that case, integrator will use next version number automatically.

Screenshot 2022-01-25 145037

## Additional info
Currently applicable only for Photoshop triggered by webpublisher with studio processing.

## Testing notes:
1. Try to publish PS workfile without version number via Webpublisher (should fail)
2. Apply this PR
3. Add 'photoshop' into `Skip for host if headless publish` in `project_settings/global/publish/CollectSceneVersion`
4. Publish it again

### CODE REVIEW

File: openpype/settings/entities/schemas/projects_schema/schemas/schema_global_publish.json [hunk 1]
```diff
@@ -18,6 +18,27 @@
...
             ]
         },
+        {
+            "type": "dict",
+            "collapsible": true,
+            "key": "CollectSceneVersion",
+            "label": "Collect Version from Workfile",
+            "is_group": true,
+            "children": [
+                {
```
Comment (iLLiCiTiT) [inline openpype/settings/entities/schemas/projects_schema/schemas/schema_global_publish.json:28]: I don't think we should give option to change `hosts`. At least they should be limited to hosts that make sense.

File: openpype/plugins/publish/collect_scene_version.py [hunk 2]
```diff
@@ -26,7 +27,20 @@ class CollectSceneVersion(pyblish.api.ContextPlugin):
         "tvpaint"
     ]
 
+    # in some cases of headless publishing (for example webpublisher using PS)
+    # you want to ignore version from name and let integrate use next version
+    skip_hosts_headless_publish = [
+
+    ]
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/collect_scene_version.py:34]:
```suggestion
    skip_hosts_headless_publish = []
```

File: openpype/plugins/publish/collect_scene_version.py [hunk 3]
```diff
@@ -26,7 +27,20 @@ class CollectSceneVersion(pyblish.api.ContextPlugin):
...
+
     def process(self, context):
+        # tests should be close to regular publish as possible
+        if (context.data["hostName"] in self.skip_hosts_headless_publish
+                and os.environ.get("HEADLESS_PUBLISH")
+                and not os.environ.get("IS_TEST")):
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/collect_scene_version.py:40]: Changed order of conditions to first check the main reason and then subconditions of the reason.
```suggestion
        if (
            os.environ.get("HEADLESS_PUBLISH")
            and not os.environ.get("IS_TEST")
            and context.data["hostName"] in self.skip_hosts_headless_publish):
```

**Review by 64qam — APPROVED** [review]
Looks good

**Pull request merged**